### PR TITLE
Workday 12.0.0 User Guide

### PR DESCRIPTION
**Pull request by dhirsch1**

### CODE REVIEW

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+** `Get Job Profiles`
+* `Recruiting` Web Service
+** `Get Evergreen Requisitions`
+** `Get Job Requisitions`
+* `Staffing` Web Service
+** `Get Workers`
+** `Get Positions`
+
+*Please make sure this wording is correct.*
+The following example shows how to use the `On Modified Objects` input source. In this example, the `On Modified Objects` input source listens for worker-related events on the `/put-worker` path and initiates the app when a new worker is added. The app then creates a new position, hires the worker, and logs information about the new hire to the Anypoint Studio console.
```
Comment (marianborzik) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: The flow does following: 
- We use te On Modified Object input source to listen all event which regards to Worksers (Creating, Modifing, Deleting).
- Then we have flow /put-worker
Use case: We open url localhost:8081/put-worker which creates a new position and then hire new worker to the created position. In a certain time (depends on fixed frequency) our source the On Modified Objects recives message that we;ve hired the new worker. The triger/listener starts second flow which logs whole information from server.

**Review by marianborzik — CHANGES_REQUESTED** [review]

**Review by marianborzik — APPROVED** [review]

File: modules/ROOT/pages/workday/workday-connector-config-topics.adoc [hunk 2]
```diff
@@ -4,7 +4,7 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../../assets/images/
 
-Configure the Global element for the connector. 
+Configure the Global element for the connector.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-config-topics.adoc]: Per our term presentation list, Mule-specific use is Global Element but global, general use is "global element."

File: modules/ROOT/pages/workday/workday-connector-config-topics.adoc [hunk 3]
```diff
@@ -13,7 +13,7 @@ Configure the Global element for the connector.
 |Connection |The connection associated with the configuration.
 |Username |Enter the username to log in with.
 |Password |Enter the corresponding password.
-|Tenant Name | Enter the Workday Tenant ID, which usually has the suffix `pt_1` appended to it, such as `acme_pt1`. 
+|Tenant Name | Enter the Workday Tenant ID, which usually has the suffix `pt_1` appended to it, such as `acme_pt1`.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-config-topics.adoc]: The two suffixes do not match:  `pt_1` appended to it, such as `acme_pt1`.

Is it pt_1 or _pt1?

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+|11.1.2 |12.0.0
+|===
+
+== Changes in This Release
+
+This release contains the following changes:
+
+* Upgraded the Workday API to 34.0
+* Added the `Student Core` operation, which enables you to invoke the Workday Student Core Web Service
+* Replaced the `On New Objects` input source with the `On Modified Objects` input source, which polls data more efficiently. For more information, see <<on-modified-objects,On Modified Objects Input Source>>.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: Per style ref:
```suggestion
* Replaced the `On New Objects` input source with the `On Modified Objects` input source, which polls data more efficiently
+
+ For more information, see <<on-modified-objects,On Modified Objects Input Source>>.
```

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+* Changed all operations so that they return headers as attributes
+* Added the ability to create response timeouts for the input source (`On Modified Objects`) and all connector operations by using the new `Response Timeout` and `Response Timeout Unit` fields.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: no period
```suggestion
* Added the ability to create response timeouts for the input source (`On Modified Objects`) and all connector operations by using the new `Response Timeout` and `Response Timeout Unit` fields
```

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+* Split the connection timeout fields into value and time unit fields. For more information, see <<changes-in-connection-types,Changes in Connection Types>>.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: ditto bullet-list format per style guide:
```suggestion
* Split the connection timeout fields into value and time unit fields
+
+ For more information, see <<changes-in-connection-types,Changes in Connection Types>>.
```

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+|Field in v11.x | Replaced By
+|Connection Idle Timeout
+|Connection Idle Timeout and Connection Idle Timeout Unit
+|Connction Timeout |Connection Timeout and Connection Timeout Unit
+|===
+
+[[on-modified-objects]]
+== On Modified Objects Input Source
+
+The `On New Objects` input source in Workday Connector version 11x polled data in a Workday resource from the beginning of the specified time interval to the present. It did this for each worker in the Workday instance, which caused large time inefficiencies when the instance contained thousands of workers.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: Can we replace "caused large time inefficiencies" with something simpler?

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+
+The new `On Modified Objects` input source, instead, always polls the latest data in the specified time interval. After one poll iteration, the input source sets the earliest moment as the previous latest moment and the latest moment as the current date and time. These changes to polling make the `On Modified Objects` input source much more efficient than the `On New Objects` input source.
+
+You can use the `On Modified Objects` input source with the following web services and operations:
+
+* `Human Resources Web Service`
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:58]: Is Web Service part of the syntax for this one? It is not for the following two....
And, if not part of syntax, need we init. cap. "web service"?

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+** `Get Job Profiles` operation
+* `Recruiting` Web Service
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:60]: See above

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+** `Get Evergreen Requisitions` operation
+** `Get Job Requisitions` operation
+* `Staffing` Web Service
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:63]: See above

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+** `Get Workers` operation
+** `Get Positions` operation
+
+The following example shows how to use the `On Modified Objects` input source. This example works as follows:
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:67]: I wonder if the example should be shown first and how it works shown after the display....

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+
+* `HTTP Listener` listens for events on the `localhost:8081/put-worker` path. When `HTTP Listener` detects an event on that path, it initiates the first flow. This flow creates a new position and hires the new worker for the created position.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: bullet list format, GLOBAL:
```suggestion
* `HTTP Listener` listens for events on the `localhost:8081/put-worker` path. 
+
+ When `HTTP Listener` detects an event on that path, it initiates the first flow. This flow creates a new position and hires the new worker for the created position.
```

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+* The `On Modified Objects` input source listens for data changes on the server. When the input source receives a message that a new worker was hired, it initiates the second flow. This flow logs the worker information to the Anypoint Studio console.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: ditto
```suggestion
* The `On Modified Objects` input source listens for data changes on the server. 
+
+ When the input source receives a message that a new worker was hired, it initiates the second flow. This flow logs the worker information to the Anypoint Studio console.
```

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc (hunk 11 above)
Comment (kbostrom) [reply modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:67]: Oh...I see that it is rather long. Hmm.

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 14]
```diff
@@ -0,0 +1,256 @@
...
+
+Paste this code into your XML editor to quickly load the flow for this example into your Mule app. If needed, change the values to reflect your environment.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:77]: Do we need this paragraph; isn't this what audiences typically do?
And wouldn't they always need to customize the values?

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+[%header%autowidth.spread]
+|===
+|Field in v11.x | Replaced By
+|Connection Idle Timeout
+|Connection Idle Timeout and Connection Idle Timeout Unit
+|Connction Timeout |Connection Timeout and Connection Timeout Unit
+|===
+
+[[on-modified-objects]]
+== On Modified Objects Input Source
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:50]: ...and Example?

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+            </scheduling-strategy>
+        </workday:modified-objects-listener>
+        <logger level="INFO" doc:name="Logger - On Modified Objects" message='#[payload]'/>
+    </flow>
+    </mule>
+----
+
+== Upgrade Prerequisites
+
+Before you perform the upgrade, you must:
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: Specify the upgrade endpoints?
```suggestion
Before you upgrade Worker Connector from version 11.x to version 12.0.0, you must:
```

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+
+[[changes-in-connection-types]]
+== Changes in Connection Types
+
+The following fields are now split into value and time unit fields:
+
+// [cols="50,50"]
+[%header%autowidth.spread]
+|===
+|Field in v11.x | Replaced By
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: 11.x or 11.1.x?

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc (hunk 16 above)
Comment (kbostrom) [reply modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]: 11.x or 11.1.x?

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 18]
```diff
@@ -0,0 +1,256 @@
...
+
+. Create a backup of your files, data, and configuration in case you need to restore to the previous version.
+. Install Workday Connector v12.0 to replace the Workday operations that were previously included in Workday Connector v11.1.x.
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade from Workday Connector v11.1.x to Workday Connector v12.0:
+
+. In Studio, create a Mule project.
+. In the Mule Palette view, click *Search in Exchange*.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc:237]: This is actually the UI label, right: *Mule Palette*

File: modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+. Check the project `pom.xml` file and verify that there are no problems.
+. Test the connection and verify that the operations work.
+
+== Troubleshooting
+
+If there are problems with caching the parameters and caching the metadata, try restarting Studio.
+
+== Revert the Upgrade
+
+If it is necessary to revert to the previous version of Workday Connector, change the `workday-connector` dependency version `12.0.0` in the project's `pom.xml` file to the previous version.
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-upgrade-migrate.adoc]:
```suggestion
If you must revert to the previous version of Workday Connector, change the `workday-connector` dependency version `12.0.0` in the project's `pom.xml` file to the previous version.
```

File: modules/ROOT/pages/workday/workday-connector-xml-maven.adoc [hunk 20]
```diff
@@ -61,11 +61,18 @@ translates the `RELEASE` dependency keyword to the current version.
 <dependency>
   <groupId>com.mulesoft.connectors</groupId>
   <artifactId>mule-workday-connector</artifactId>
-  <version>RELEASE</version>
+  <version>12.0.0</version>
   <classifier>mule-plugin</classifier>
 </dependency>
 ----
 
+To obtain the most up-to-date pom.xml file information:
```
Comment (kbostrom) [inline modules/ROOT/pages/workday/workday-connector-xml-maven.adoc]:
```suggestion
To obtain the most up-to-date `pom.xml` file information:
```

**Review by kbostrom — APPROVED** [review]
Edited

**Pull request merged**